diff --git a/CMakeLists.txt b/CMakeLists.txt
index e637b86..6a61500 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,7 +92,7 @@ IF ( NOT VERSION )
    message( STATUS "version remainder: ${VERSION-REMAINDER}" ) 
    # SET(BUILD_NUMBER ${GIT_VERSION})
    SET(BUILD_NUMBER ${VERSION-BASE}) 
-   SET(VERSION ${MAJOR_VERSION}.${MINOR_VERSION}.${BUILD_NUMBER}-${VERSION-REMAINDER}) 
+   SET(VERSION "1.3.0-0") 
 ENDIF()
 message( STATUS "Software Version: ${VERSION}" )
 
