diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac78173..2523692 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,6 +89,7 @@ IF ( NOT VERSION )
    message( STATUS "version remainder: ${VERSION-REMAINDER}" )
    SET(BUILD_NUMBER ${VERSION-BASE})
    SET(VERSION ${MAJOR_VERSION}.${MINOR_VERSION}-${GIT_RELEASE_COMMITS}) #-${VERSION-REMAINDER})
+   SET(VERSION "2.4-0")
 ENDIF()
 message( STATUS "Software Version: ${VERSION}" )
 
