From 25e180462e0bc00a3781056b66b17732085dcf70 Mon Sep 17 00:00:00 2001
From: "tristan.ramseyer" <tristan.ramseyer@stromerbike.com>
Date: Tue, 3 Sep 2024 14:51:48 +0200
Subject: [PATCH] print warning as info in gpiolib-sysfs.c

Upstream-Status: Inappropriate [stromer-specific]
Signed-off-by: tristan.ramseyer <tristan.ramseyer@stromerbike.com>
---
 drivers/gpio/gpiolib-sysfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/gpiolib-sysfs.c b/drivers/gpio/gpiolib-sysfs.c
index 44c1ad51b3fe..50cf4ff522ad 100644
--- a/drivers/gpio/gpiolib-sysfs.c
+++ b/drivers/gpio/gpiolib-sysfs.c
@@ -460,7 +460,7 @@ static ssize_t export_store(struct class *class,
 	desc = gpio_to_desc(gpio);
 	/* reject invalid GPIOs */
 	if (!desc) {
-		pr_warn("%s: invalid GPIO %ld\n", __func__, gpio);
+		pr_info("%s: invalid GPIO %ld\n", __func__, gpio);
 		return -EINVAL;
 	}
 	gc = desc->gdev->chip;
@@ -510,7 +510,7 @@ static ssize_t unexport_store(struct class *class,
 	desc = gpio_to_desc(gpio);
 	/* reject bogus commands (gpio_unexport ignores them) */
 	if (!desc) {
-		pr_warn("%s: invalid GPIO %ld\n", __func__, gpio);
+		pr_info("%s: invalid GPIO %ld\n", __func__, gpio);
 		return -EINVAL;
 	}
 
-- 
2.34.1

