diff --git a/drivers/gpio/gpiolib-sysfs.c b/drivers/gpio/gpiolib-sysfs.c
index 44c1ad51b3fe..50cf4ff522ad 100644
--- a/drivers/gpio/gpiolib-sysfs.c
+++ b/drivers/gpio/gpiolib-sysfs.c
@@ -460,7 +460,7 @@ static ssize_t export_store(struct class *class,
 	desc = gpio_to_desc(gpio);
 	/* reject invalid GPIOs */
 	if (!desc) {
-		pr_warn("%s: invalid GPIO %ld\n", __func__, gpio);
+		pr_info("%s: invalid GPIO %ld\n", __func__, gpio);
 		return -EINVAL;
 	}
 	gc = desc->gdev->chip;
@@ -510,7 +510,7 @@ static ssize_t unexport_store(struct class *class,
 	desc = gpio_to_desc(gpio);
 	/* reject bogus commands (gpio_unexport ignores them) */
 	if (!desc) {
-		pr_warn("%s: invalid GPIO %ld\n", __func__, gpio);
+		pr_info("%s: invalid GPIO %ld\n", __func__, gpio);
 		return -EINVAL;
 	}
 
