diff --git a/arch/arm/boot/dts/imx6ul-phytec-phyboard-segin-ff-rdk.dts b/arch/arm/boot/dts/imx6ul-phytec-phyboard-segin-ff-rdk.dts
index 5d2482f..7ea667d 100644
--- a/arch/arm/boot/dts/imx6ul-phytec-phyboard-segin-ff-rdk.dts
+++ b/arch/arm/boot/dts/imx6ul-phytec-phyboard-segin-ff-rdk.dts
@@ -48,10 +48,6 @@
 	status = "okay";
 };
 
-&sai2 {
-	status = "okay";
-};
-
 &sound {
 	status = "okay";
 };
