From e3a55b2fe61cca64f949b5167b64330d45229124 Mon Sep 17 00:00:00 2001
From: "tristan.ramseyer" <tristan.ramseyer@stromerbike.com>
Date: Mon, 2 Sep 2024 23:10:04 +0200
Subject: [PATCH] systemd-udev-trigger.service

Upstream-Status: Inappropriate [stromer-specific]
Signed-off-by: tristan.ramseyer <tristan.ramseyer@stromerbike.com>
---
 units/systemd-udev-trigger.service | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/units/systemd-udev-trigger.service b/units/systemd-udev-trigger.service
index cb1e4f9fea..3fa26c3240 100644
--- a/units/systemd-udev-trigger.service
+++ b/units/systemd-udev-trigger.service
@@ -8,15 +8,14 @@
 #  (at your option) any later version.
 
 [Unit]
-Description=Coldplug All udev Devices
+Description=Coldplug all Devices except input and ubi
 Documentation=man:udev(7) man:systemd-udevd.service(8)
 DefaultDependencies=no
 Wants=systemd-udevd.service
 After=systemd-udevd-kernel.socket systemd-udevd-control.socket
-Before=sysinit.target
 ConditionPathIsReadWrite=/sys
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=-udevadm trigger --type=all --action=add --prioritized-subsystem=module,block,tpmrm,net,tty,input
+ExecStart=-udevadm trigger -S input -S ubi --type=all --action=add --prioritized-subsystem=module,block,tpmrm,net,tty,input
-- 
2.34.1

