diff --git a/src/basic/time-util.c b/src/basic/time-util.c
index b659d6905d..b8591b56b2 100644
--- a/src/basic/time-util.c
+++ b/src/basic/time-util.c
@@ -11,6 +11,7 @@
 #include <unistd.h>
 
 #include "alloc-util.h"
+#include "chase-symlinks.h"
 #include "fd-util.h"
 #include "fileio.h"
 #include "fs-util.h"
@@ -1497,7 +1498,7 @@ int get_timezone(char **ret) {
         char *z;
         int r;
 
-        r = readlink_malloc("/etc/localtime", &t);
+        r = chase_symlinks("/etc/localtime", NULL, 0, &t, NULL);
         if (r == -ENOENT) {
                 /* If the symlink does not exist, assume "UTC", like glibc does */
                 z = strdup("UTC");
