diff --git a/units/systemd-timesyncd.service.in b/units/systemd-timesyncd.service.in
index d3bc4e9841..96873bde0a 100644
--- a/units/systemd-timesyncd.service.in
+++ b/units/systemd-timesyncd.service.in
@@ -17,6 +17,7 @@ After=systemd-remount-fs.service systemd-sysusers.service
 Before=time-sync.target sysinit.target shutdown.target
 Conflicts=shutdown.target
 Wants=time-sync.target
+RequiresMountsFor=/mnt/data
 
 [Service]
 Type=notify
@@ -24,10 +25,6 @@ Restart=always
 RestartSec=0
 ExecStart=!!@rootlibexecdir@/systemd-timesyncd
 WatchdogSec=3min
-User=systemd-timesync
-DynamicUser=yes
-CapabilityBoundingSet=CAP_SYS_TIME
-AmbientCapabilities=CAP_SYS_TIME
 PrivateDevices=yes
 ProtectHome=yes
 ProtectControlGroups=yes
@@ -40,7 +37,6 @@ RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
 SystemCallFilter=~@cpu-emulation @debug @keyring @module @mount @obsolete @raw-io @reboot @swap
 SystemCallArchitectures=native
 LockPersonality=yes
-StateDirectory=systemd/timesync
 
 [Install]
 WantedBy=sysinit.target
