diff --git a/src/timesync/timesyncd-manager.c b/src/timesync/timesyncd-manager.c
index 3281d09c76..7b0e5f0ba1 100644
--- a/src/timesync/timesyncd-manager.c
+++ b/src/timesync/timesyncd-manager.c
@@ -368,6 +368,7 @@ static int manager_adjust_clock(Manager *m, double offset, int leap_sec) {
 
         /* If touch fails, there isn't much we can do. Maybe it'll work next time. */
         (void) touch("/mnt/data/timesync/clock");
+        (void) touch("/tmp/synchronized");
 
         m->drift_ppm = tmx.freq / 65536;
 
