diff --git a/units/systemd-resolved.service.in b/units/systemd-resolved.service.in
index ef5398cbf..fbfafc9ad 100644
--- a/units/systemd-resolved.service.in
+++ b/units/systemd-resolved.service.in
@@ -14,8 +14,8 @@ Documentation=https://www.freedesktop.org/wiki/Software/systemd/resolved
 Documentation=https://www.freedesktop.org/wiki/Software/systemd/writing-network-configuration-managers
 Documentation=https://www.freedesktop.org/wiki/Software/systemd/writing-resolver-clients
 DefaultDependencies=no
-After=systemd-sysusers.service systemd-networkd.service
-Before=network.target nss-lookup.target shutdown.target
+After=systemd-sysusers.service systemd-networkd.service check.target
+Before=shutdown.target
 Conflicts=shutdown.target
 Wants=nss-lookup.target
 
@@ -47,5 +47,5 @@ RuntimeDirectory=systemd/resolve
 RuntimeDirectoryPreserve=yes
 
 [Install]
-WantedBy=multi-user.target
+WantedBy=communication.target
 Alias=dbus-org.freedesktop.resolve1.service
