diff --git a/units/systemd-timesyncd.service.in b/units/systemd-timesyncd.service.in
index c606461091..f3842cbb67 100644
--- a/units/systemd-timesyncd.service.in
+++ b/units/systemd-timesyncd.service.in
@@ -17,6 +17,7 @@ After=systemd-sysusers.service
 Before=time-set.target sysinit.target shutdown.target
 Conflicts=shutdown.target
 Wants=time-set.target
+RequiresMountsFor=/mnt/data
 
 [Service]
 AmbientCapabilities=CAP_SYS_TIME
@@ -40,6 +41,7 @@ ProtectKernelLogs=yes
 ProtectKernelModules=yes
 ProtectKernelTunables=yes
 ProtectSystem=strict
+ReadWritePaths=/mnt/data/timesync
 Restart=always
 RestartSec=0
 RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
