diff --git a/units/systemd-timesyncd.service.in b/units/systemd-timesyncd.service.in
index c606461091..3090a28147 100644
--- a/units/systemd-timesyncd.service.in
+++ b/units/systemd-timesyncd.service.in
@@ -17,11 +17,10 @@ After=systemd-sysusers.service
 Before=time-set.target sysinit.target shutdown.target
 Conflicts=shutdown.target
 Wants=time-set.target
+RequiresMountsFor=/mnt/data
 
 [Service]
-AmbientCapabilities=CAP_SYS_TIME
 BusName=org.freedesktop.timesync1
-CapabilityBoundingSet=CAP_SYS_TIME
 # Turn off DNSSEC validation for hostname look-ups, since those need the
 # correct time to work, but we likely won't acquire that without NTP. Let's
 # break this chicken-and-egg cycle here.
@@ -47,12 +46,10 @@ RestrictNamespaces=yes
 RestrictRealtime=yes
 RestrictSUIDSGID=yes
 RuntimeDirectory=systemd/timesync
-StateDirectory=systemd/timesync
 SystemCallArchitectures=native
 SystemCallErrorNumber=EPERM
 SystemCallFilter=@system-service @clock
 Type=notify
-User=systemd-timesync
 {{SERVICE_WATCHDOG}}
 
 [Install]
