diff --git a/units/systemd-journal-upload.service.in b/units/systemd-journal-upload.service.in
index 42da70f473..4464b66822 100644
--- a/units/systemd-journal-upload.service.in
+++ b/units/systemd-journal-upload.service.in
@@ -10,8 +10,6 @@
 [Unit]
 Description=Journal Remote Upload Service
 Documentation=man:systemd-journal-upload(8)
-Wants=network-online.target
-After=network-online.target
 
 [Service]
 ExecStart=@rootlibexecdir@/systemd-journal-upload --save-state

