diff --git a/units/systemd-journal-upload.service.in b/units/systemd-journal-upload.service.in
index e3800473ec..a70315479c 100644
--- a/units/systemd-journal-upload.service.in
+++ b/units/systemd-journal-upload.service.in
@@ -10,8 +10,6 @@
 [Unit]
 Description=Journal Remote Upload Service
 Documentation=man:systemd-journal-upload(8)
-Wants=network-online.target
-After=network-online.target
 
 [Service]
 DynamicUser=yes

