From 11455e247c8cf774eaba9764b23ea4ebcb37b46e Mon Sep 17 00:00:00 2001
From: "tristan.ramseyer" <tristan.ramseyer@stromerbike.com>
Date: Mon, 2 Sep 2024 19:10:46 +0200
Subject: [PATCH] remove desktop file and icon

Upstream-Status: Inappropriate [stromer-specific]
Signed-off-by: tristan.ramseyer <tristan.ramseyer@stromerbike.com>
---
 Makefile.am | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 2a9cc29f..8d68cfe6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,19 +10,10 @@ dist_man_MANS = htop.1
 EXTRA_DIST = \
 	$(dist_man_MANS) \
 	autogen.sh \
-	htop.desktop \
-	htop.png \
-	htop.svg \
 	build-aux/compile \
 	build-aux/depcomp \
 	build-aux/install-sh \
 	build-aux/missing
-applicationsdir = $(datadir)/applications
-applications_DATA = htop.desktop
-pixmapdir = $(datadir)/pixmaps
-pixmap_DATA = htop.png
-appicondir = $(datadir)/icons/hicolor/scalable/apps
-appicon_DATA = htop.svg
 
 AM_CFLAGS += -pedantic -std=c99 -D_XOPEN_SOURCE_EXTENDED -DSYSCONFDIR="\"$(sysconfdir)\"" -I"$(top_srcdir)/$(my_htop_platform)"
 AM_LDFLAGS =
-- 
2.34.1

