diff --git a/Makefile.am b/Makefile.am
index e033c35..e73087f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -5,12 +5,8 @@ AUTOMAKE_OPTIONS = subdir-objects
 bin_PROGRAMS = htop
 
 dist_man_MANS = htop.1
-EXTRA_DIST = $(dist_man_MANS) htop.desktop htop.png scripts/MakeHeader.py \
+EXTRA_DIST = $(dist_man_MANS) scripts/MakeHeader.py \
 install-sh autogen.sh missing
-applicationsdir = $(datadir)/applications
-applications_DATA = htop.desktop
-pixmapdir = $(datadir)/pixmaps
-pixmap_DATA = htop.png
 
 htop_CFLAGS = -pedantic -Wall $(wextra_flag) -std=c99 -D_XOPEN_SOURCE_EXTENDED -DSYSCONFDIR=\"$(sysconfdir)\" -I"$(top_srcdir)/$(my_htop_platform)"
 htop_LDFLAGS = 

