From 2487c1c82f4e7ad46547cf5f486980ebcd4b17ed Mon Sep 17 00:00:00 2001
From: "tristan.ramseyer" <tristan.ramseyer@stromerbike.com>
Date: Mon, 2 Sep 2024 18:57:35 +0200
Subject: [PATCH] ldattach adjust gsm config

Upstream-Status: Inappropriate [stromer-specific]
Signed-off-by: tristan.ramseyer <tristan.ramseyer@stromerbike.com>
---
 sys-utils/ldattach.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sys-utils/ldattach.c b/sys-utils/ldattach.c
index 9c67eeec8..1b2b4f48b 100644
--- a/sys-utils/ldattach.c
+++ b/sys-utils/ldattach.c
@@ -270,9 +270,9 @@ static void gsm0710_set_conf(int tty_fd)
 	/*  we are initiator and need encoding 0 (basic) */
 	c.initiator = 1;
 	c.encapsulation = 0;
-	/*  our modem defaults to a maximum size of 127 bytes */
-	c.mru = 127;
-	c.mtu = 127;
+	/*  our modem defaults to a maximum size of 31 bytes */
+	c.mru = 31;
+	c.mtu = 31;
 	/*  set the new configuration */
 	ioctl(tty_fd, GSMIOC_SETCONF, &c);
 	/* Add by guowenxue end*/
-- 
2.34.1

