diff --git a/sys-utils/ldattach.c b/sys-utils/ldattach.c
index d33d68535..3335b765c 100644
--- a/sys-utils/ldattach.c
+++ b/sys-utils/ldattach.c
@@ -270,9 +270,9 @@ static void gsm0710_set_conf(int tty_fd)
 	/*  we are initiator and need encoding 0 (basic) */
 	c.initiator = 1;
 	c.encapsulation = 0;
-	/*  our modem defaults to a maximum size of 127 bytes */
-	c.mru = 127;
-	c.mtu = 127;
+	/*  our modem defaults to a maximum size of 31 bytes */
+	c.mru = 31;
+	c.mtu = 31;
 	/*  set the new configuration */
 	ioctl(tty_fd, GSMIOC_SETCONF, &c);
 	/* Add by guowenxue end*/

