diff --git a/src/log_format_loader.cc b/src/log_format_loader.cc
index 7c131df..917a695 100644
--- a/src/log_format_loader.cc
+++ b/src/log_format_loader.cc
@@ -665,7 +665,6 @@ static void write_sample_file(void)
         (write(sample_fd.get(),
                default_log_formats_json,
                strlen(default_log_formats_json)) == -1)) {
-        perror("error: unable to write default format file");
     }
 
     string sh_path = dotlnav_path("formats/default/dump-pid.sh");
@@ -673,7 +672,6 @@ static void write_sample_file(void)
 
     if ((sh_fd = open(sh_path.c_str(), O_WRONLY|O_TRUNC|O_CREAT, 0755)) == -1 ||
         write(sh_fd.get(), dump_pid_sh, strlen(dump_pid_sh)) == -1) {
-        perror("error: unable to write default text file");
     }
 
     static const char *SCRIPTS[] = {
@@ -703,7 +701,6 @@ static void write_sample_file(void)
         }
         if ((script_fd = open(script_path.c_str(), O_WRONLY|O_TRUNC|O_CREAT, 0755)) == -1 ||
             write(script_fd.get(), script_content, script_len) == -1) {
-            perror("error: unable to write default text file");
         }
     }
 }
diff --git a/src/session_data.cc b/src/session_data.cc
index b07e964..13c40d4 100644
--- a/src/session_data.cc
+++ b/src/session_data.cc
@@ -1217,7 +1217,6 @@ void save_session()
     view_file_tmp_name = view_file_name + ".tmp";
 
     if ((file = fopen(view_file_tmp_name.c_str(), "w")) == nullptr) {
-        perror("Unable to open session file");
     }
     else if (nullptr == (handle = yajl_gen_alloc(nullptr))) {
         perror("Unable to create yajl_gen object");

