diff --git a/configure.ac b/configure.ac
index f36438b..63f35c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,23 +10,6 @@ AC_CANONICAL_HOST
 
 AX_CXX_COMPILE_STDCXX_14([noext], [mandatory])
 
-for defdir in /opt/local /usr/local /usr /; do
-    if test -d "$defdir/include"; then
-        CPPFLAGS="$CPPFLAGS -I$defdir/include"
-    fi
-
-    if test -d "$defdir/lib64"; then
-        LDFLAGS="$LDFLAGS -L$defdir/lib64"
-    fi
-    if test -d "$defdir/lib"; then
-        LDFLAGS="$LDFLAGS -L$defdir/lib"
-    fi
-
-    if test -d "$defdir/lib/x86_64-linux-gnu"; then
-        LDFLAGS="$LDFLAGS -L$defdir/lib/x86_64-linux-gnu"
-    fi
-done
-
 dnl abssrcdir is the absolute path to the source base (regardless of where
 dnl you are building it)
 AS_CASE([x$srcdir],
diff --git a/src/Makefile.am b/src/Makefile.am
index 6b38a15..de3ab7c 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,7 +1,7 @@
 
 bin_PROGRAMS = lnav
 
-noinst_PROGRAMS = bin2c ptimec lnav-test
+noinst_PROGRAMS = lnav-test
 
 noinst_LIBRARIES = libdiag.a
 
@@ -17,6 +17,12 @@ RE2C_V = $(RE2C_V_@AM_V@)
 RE2C_V_ = $(RE2C_V_@AM_DEFAULT_V@)
 RE2C_V_0 = @echo "  RE2C   " $@;
 
+ptimec: $(srcdir)/ptimec.cc
+	gcc $(srcdir)/ptimec.cc -o ptimec
+
+bin2c: $(srcdir)/bin2c.c
+	gcc $(srcdir)/bin2c.c -o bin2c
+
 help.c: $(srcdir)/help.txt bin2c
 	$(BIN2C_V)./bin2c -z -c $(srcdir)/help.txt $@
 
@@ -381,12 +387,6 @@ lnav_test_LDADD = \
 	$(TEXT2C_FILES) \
 	$(LDADD)
 
-bin2c_SOURCES = bin2c.c
-bin2c_LDADD =
-
-ptimec_SOURCES = ptimec.cc
-ptimec_LDADD =
-
 DISTCLEANFILES = \
 	data_scanner_re.cc \
 	dump-pid-sh.c \
