From 9818fa7f7b304ccfc8f1b195dfac0e1e2c26d463 Mon Sep 17 00:00:00 2001
From: "tristan.ramseyer" <tristan.ramseyer@stromerbike.com>
Date: Mon, 2 Sep 2024 18:08:08 +0200
Subject: [PATCH] add libusb directory configure arg

Upstream-Status: Inappropriate [yocto]
Signed-off-by: tristan.ramseyer <tristan.ramseyer@stromerbike.com>
---
 configure.ac | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5cd15067b..097fef5ab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -914,19 +914,19 @@ if test x"$LIBUSB_NAME" != x ; then
                   have_libusb=yes ])
    AC_MSG_CHECKING([libusb include dir])
    usb_incdir_found="no"
-   for _incdir in "" "/usr/include/libusb-1.0" \
-       "/usr/local/include/libusb-1.0" "/usr/pkg/include/libusb-1.0"; do
-     _libusb_save_cppflags=$CPPFLAGS
-     if test -n "${_incdir}"; then
-       CPPFLAGS="-I${_incdir} ${CPPFLAGS}"
-     fi
-     AC_PREPROC_IFELSE([AC_LANG_SOURCE([[@%:@include <libusb.h>]])],
-     [usb_incdir=${_incdir}; usb_incdir_found="yes"], [])
-     CPPFLAGS=${_libusb_save_cppflags}
-     if test "$usb_incdir_found" = "yes"; then
-       break
-     fi
-   done
+   AC_ARG_WITH(libusb,
+     [  --with-libusb=DIR  use libusb in DIR],[
+       if test -d "$withval"; then
+         _incdir="${withval}/libusb-1.0"
+         _libusb_save_cppflags=$CPPFLAGS
+         if test -n "${_incdir}"; then
+           CPPFLAGS="-I${_incdir} ${CPPFLAGS}"
+         fi
+         AC_PREPROC_IFELSE([AC_LANG_SOURCE([[@%:@include <libusb.h>]])],
+         [usb_incdir=${_incdir}; usb_incdir_found="yes"], [])
+         CPPFLAGS=${_libusb_save_cppflags}
+       fi
+     ])
    if test "$usb_incdir_found" = "yes"; then
      AC_MSG_RESULT([${usb_incdir}])
    else
-- 
2.34.1

