diff --git a/configure.ac b/configure.ac
index 382ef1df5..5ee2defe5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -820,18 +820,19 @@ if test x"$LIBUSB_NAME" != x ; then
                   have_libusb=yes ])
    AC_MSG_CHECKING([libusb include dir])
    usb_incdir_found="no"
-   for _incdir in "" "/usr/include/libusb-1.0" "/usr/local/include/libusb-1.0"; do
-     _libusb_save_cppflags=$CPPFLAGS
-     if test -n "${_incdir}"; then
-       CPPFLAGS="-I${_incdir} ${CPPFLAGS}"
-     fi
-     AC_PREPROC_IFELSE([AC_LANG_SOURCE([[@%:@include <libusb.h>]])],
-     [usb_incdir=${_incdir}; usb_incdir_found="yes"], [])
-     CPPFLAGS=${_libusb_save_cppflags}
-     if test "$usb_incdir_found" = "yes"; then
-       break
-     fi
-   done
+   AC_ARG_WITH(libusb,
+     [  --with-libusb=DIR  use libusb in DIR],[
+       if test -d "$withval"; then
+         _incdir="${withval}/libusb-1.0"
+         _libusb_save_cppflags=$CPPFLAGS
+         if test -n "${_incdir}"; then
+           CPPFLAGS="-I${_incdir} ${CPPFLAGS}"
+         fi
+         AC_PREPROC_IFELSE([AC_LANG_SOURCE([[@%:@include <libusb.h>]])],
+         [usb_incdir=${_incdir}; usb_incdir_found="yes"], [])
+         CPPFLAGS=${_libusb_save_cppflags}
+       fi
+     ])
    if test "$usb_incdir_found" = "yes"; then
      AC_MSG_RESULT([${usb_incdir}])
    else

