diff --git a/src/bluetooth.service.in b/src/bluetooth.service.in
index f9faaa452..184d0dee2 100644
--- a/src/bluetooth.service.in
+++ b/src/bluetooth.service.in
@@ -2,11 +2,12 @@
 Description=Bluetooth service
 Documentation=man:bluetoothd(8)
 ConditionPathIsDirectory=/sys/class/bluetooth
+RequiresMountsFor=/mnt/data
 
 [Service]
 Type=dbus
 BusName=org.bluez
-ExecStart=@pkglibexecdir@/bluetoothd
+ExecStart=@pkglibexecdir@/bluetoothd --noplugin=hostname
 NotifyAccess=main
 #WatchdogSec=10
 #Restart=on-failure

