diff --git a/src/bluetooth.service.in b/src/bluetooth.service.in
index beb98ce0c..389d5ad35 100644
--- a/src/bluetooth.service.in
+++ b/src/bluetooth.service.in
@@ -2,11 +2,12 @@
 Description=Bluetooth service
 Documentation=man:bluetoothd(8)
 ConditionPathIsDirectory=/sys/class/bluetooth
+RequiresMountsFor=/mnt/data
 
 [Service]
 Type=dbus
 BusName=org.bluez
-ExecStart=@pkglibexecdir@/bluetoothd
+ExecStart=@pkglibexecdir@/bluetoothd --noplugin=hostname
 NotifyAccess=main
 #WatchdogSec=10
 #Restart=on-failure
@@ -22,7 +23,7 @@ ProtectControlGroups=true
 StateDirectory=bluetooth
 StateDirectoryMode=0700
 ConfigurationDirectory=bluetooth
-ConfigurationDirectoryMode=0555
+ConfigurationDirectoryMode=0755
 
 # Execute Mappings
 MemoryDenyWriteExecute=true
