diff --git a/rsyslog.service.in b/rsyslog.service.in
index b761ae571..60dd34278 100644
--- a/rsyslog.service.in
+++ b/rsyslog.service.in
@@ -1,11 +1,13 @@
 [Unit]
 Description=System Logging Service
 Requires=syslog.socket
+Before=medusa-DataServer.service
 Documentation=man:rsyslogd(8)
 Documentation=https://www.rsyslog.com/doc/
 
 [Service]
 Type=notify
+ExecStartPre=/bin/mkdir -p /mnt/data/rsyslog
 ExecStart=@sbindir@/rsyslogd -n -iNONE
 StandardOutput=null
 Restart=on-failure
@@ -15,5 +17,5 @@ Restart=on-failure
 LimitNOFILE=16384
 
 [Install]
-WantedBy=multi-user.target
+WantedBy=drive.target
 Alias=syslog.service
